### PR TITLE
Support Eltwise3D on OCV backend 

### PR DESCRIPTION
**Pull request by l-bat**
**Merge with extra**: https://github.com/opencv/opencv_extra/pull/621

```
force_builders=Custom,Custom Win,Custom Mac
build_image:Custom=ubuntu-openvino-2019r1:16.04
build_image:Custom Win=openvino-2019r1
build_image:Custom Mac=openvino-2019r1

test_modules:Custom=dnn,python2,python3,java
test_modules:Custom Win=dnn,python2,python3,java
test_modules:Custom Mac=dnn,python2,python3,java

buildworker:Custom=linux-1
test_opencl:Custom=ON
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

File: modules/dnn/src/layers/eltwise_layer.cpp [hunk 1]
```diff
@@ -156,9 +156,15 @@ class EltwiseLayerImpl CV_FINAL : public EltwiseLayer
...
-            p.channels = (dst.dims == 4 ? dst.size[1] : 1);
-            p.planeSize = (dst.dims >= 3 ? dst.size[dst.dims - 1] * dst.size[dst.dims - 2] :
-                                           dst.size[dst.dims - 1]);
+            p.channels = (dst.dims >= 4 ? dst.size[1] : 1);
+
+            switch (dst.dims) {
+                case 3:  p.planeSize = dst.size[1] * dst.size[2];               break;
+                case 4:  p.planeSize = dst.size[2] * dst.size[3];               break;
+                case 5:  p.planeSize = dst.size[2] * dst.size[3] * dst.size[4]; break;
+                default: p.planeSize = dst.size[dst.dims - 1];
```
Comment (dkurt) [inline modules/dnn/src/layers/eltwise_layer.cpp]: What about `total` method from https://github.com/opencv/opencv/blob/master/modules/dnn/include/opencv2/dnn/shape_utils.hpp? We just need to multiply all the dimensions starts from 2.

Comment (l-bat) [reply modules/dnn/src/layers/eltwise_layer.cpp]: if `dst.dims = 3` we multiply 1 and 2 dimensions.

Comment (l-bat) [reply modules/dnn/src/layers/eltwise_layer.cpp]: Maybe `p.planeSize = dst.total(1) / p.channels`?

Comment (dkurt) [reply modules/dnn/src/layers/eltwise_layer.cpp]: @l-bat, I think that `p.planeSize = dst.total(dst.dims >= 4 ? 2 : 1)` is better.

**Comment by dkurt** [comment]
@l-bat, Please check IE tests: https://pullrequest.opencv.org/buildbot/builders/precommit_custom_linux/builds/2246

```
[ RUN      ] Test_ONNX_layers.Eltwise3D/2, where GetParam() = DLIE/OCL
unknown file: Failure
C++ exception with description "OpenCV(3.4.6-dev) /build/precommit_custom_linux/3.4/opencv/modules/dnn/src/op_inf_engine.cpp:450: error: (-215:Assertion failed) Failed to initialize Inference Engine backend: Invalid data dimensions in function 'initPlugin'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Eltwise3D/2, where GetParam() = DLIE/OCL (56 ms)
```

**Pull request merged**